querier/src/cache_system/backend/addressable_heap.rs

```diff
@@ -383,4 +385,135 @@ mod tests {
         assert_eq!(heap.insert(2, "b", 3), None);
         assert_eq!(heap.insert(1, "c", 5), Some(("a", 4)));
     }
+
+    /// Simple version of [`AddressableHeap`] for testing.
+    struct SimpleAddressableHeap {
+        inner: Vec<(u8, String, i8)>,
+    }
+
+    impl SimpleAddressableHeap {
+        fn new() -> Self {
+            Self { inner: Vec::new() }
+        }
+
+        fn insert(&mut self, k: u8, v: String, o: i8) -> Option<(String, i8)> {
+            let res = match self
+                .inner
+                .iter()
+                .enumerate()
+                .find(|(_idx, (k2, _v, _o))| *k2 == k)
+                .map(|(idx, _)| idx)
+            {
+                Some(idx) => {
+                    let (_k, v, o) = self.inner.remove(idx);
+                    Some((v, o))
+                }
+                None => None,
+            };
```

Comment:
Could you just call `self.remove` here?

```suggestion
            let res = self.remove(&k);
```